### ISSUE TITLE
piped request body is only sent on first request with followAllRedirects: true

### ISSUE THREAD

**Issue by kevinburkenotion** [request/request#3138]
If I make a POST request and have `followAllRedirects: true` and I pipe the request body to `request`, the request body will only be uploaded as part of the first request.

I'll submit some tests shortly, but this should be enough to verify, add this at the top of test-redirect.js, basically just a self redirecting server.

```javascript
s.on('/loop', function (req, res) {
  if (req.method === 'POST') {
    req.on('data', function(data) {
      console.log('post data', req.url, data.toString())
    })
  }
  const u = new url.URL(req.url, 'http://localhost')
  let count = 0;
  let stopAfter = 100;
  if (!u.searchParams.has('count')) {
      u.searchParams.set('count', '1')
  } else {
      count = parseInt(u.searchParams.get('count'), 10)
      u.searchParams.set('count', (count + 1).toString())
  }
  if (u.searchParams.has('stopAfter')) {
      stopAfter = parseInt(u.searchParams.get('stopAfter'), 10)
  }
  if (count >= stopAfter) {
      res.writeHead(200)
      res.end(count.toString())
      return
  }
  res.writeHead(302, {
    location: s.url + u.pathname + '?' + u.searchParams.toString()
  })
  res.end(count.toString())
})

```

Then add this test. I expect the POST body to be uploaded with every request but instead the on('data') callback in the server handler only fires once.

```javascript
tape('piped request body with max redirects', function (t) {
  t.plan(2)
  var w = new stream.PassThrough()
  w.write('test data')
  let r = request({
    uri: s.url + '/loop?stopAfter=12',
    method: 'POST',
    maxRedirects: 10,
    followAllRedirects: true,
  })
  w.pipe(r)
  r.on('error', function(err) {
    t.ok(err.message, "Exceeded maxRedirects. Probably stuck in a redirect loop ")
  })
  r.on('response', function(response) {
    t.equal(response.statusCode, 302)
  })
})
```

Basically I was confused about the `EventEmitter: memory leak detected` warning, where we continually add `.on('pipe')` handlers, and I wanted to try to actually trigger it. It turns out you can't trigger it because the POST body is only uploaded once, before the first request. As far as I can tell if you were to try to attempt to pipe it to the 2nd and 3rd requests, you'd have to unregister the first `.on('pipe')` listener or they would both run for request 2, the 1st, 2nd, and 3rd listeners would run for request 3, and so forth.
